File: src/MarkdownEditor.java
import javax.swing.JFrame;
import javax.swing.JTextArea;
import javax.swing.JButton;
import javax.swing.JScrollPane;
import javax.swing.GroupLayout;
import javax.swing.UIManager;
import javax.swing.UnsupportedLookAndFeelException;
import javax.swing.JFileChooser;
import javax.swing.BorderFactory;
import javax.swing.filechooser.FileNameExtensionFilter;

import java.awt.event.ActionEvent;
import java.awt.event.ActionListener;
import java.awt.Color;

import java.io.File;

public class MarkdownEditor extends JFrame{

    private final JTextArea textArea;
    private final JScrollPane scroll;
    private final JButton importButton;
    private final JButton saveButton;
    private final GroupLayout layout;
    private File file;

    public MarkdownEditor(){
        try {
            UIManager.setLookAndFeel(UIManager.getSystemLookAndFeelClassName());
        } catch (ClassNotFoundException | IllegalAccessException | UnsupportedLookAndFeelException |
                 InstantiationException e) {
            throw new RuntimeException(e);
        }

        this.textArea = new JTextArea("(Markdown text will appear here)");
        this.scroll = new JScrollPane(this.textArea);
        this.importButton = new JButton("Import File");
        this.saveButton = new JButton("Save Changes");
        this.layout = new GroupLayout(this.getContentPane());

        layout.setHorizontalGroup(layout.createSequentialGroup()
                .addComponent(scroll)
                .addGroup(layout.createParallelGroup(GroupLayout.Alignment.LEADING)
                        .addComponent(importButton, 200, GroupLayout.DEFAULT_SIZE, GroupLayout.PREFERRED_SIZE)
                        .addComponent(saveButton, 200, GroupLayout.DEFAULT_SIZE, GroupLayout.PREFERRED_SIZE)));

        layout.setVerticalGroup(layout.createParallelGroup(GroupLayout.Alignment.LEADING)
                .addComponent(scroll)
                .addGroup(layout.createSequentialGroup()
                        .addComponent(importButton)
                        .addComponent(saveButton)));

        importButton.addActionListener(new ActionListener() {
            @Override
            public void actionPerformed(ActionEvent e) {
                JFileChooser fc = new JFileChooser();
                fc.setFileFilter(new FileNameExtensionFilter("Markdown Files (*.md)", "md"));
                fc.showOpenDialog(null);

                file = fc.getSelectedFile();
                FileOptionsKt.loadFile();

                importButton.setEnabled(false);
                saveButton.setEnabled(true);
                textArea.setEnabled(true);
            }
        });

        saveButton.addActionListener(new ActionListener() {
            @Override
            public void actionPerformed(ActionEvent e) {
                FileOptionsKt.saveFile();
                importButton.setEnabled(true);
                saveButton.setEnabled(false);
                textArea.setText("(Markdown text will appear here)");
                textArea.setEnabled(false);
            }
        });

        scroll.setBorder(BorderFactory.createLineBorder(Color.BLACK));
        textArea.setEnabled(false);
        textArea.setLineWrap(true);
        saveButton.setEnabled(false);
        layout.setAutoCreateGaps(true);
        layout.setAutoCreateContainerGaps(true);

        setLayout(layout);
        setTitle("MarkdownEditor");
        setDefaultCloseOperation(JFrame.EXIT_ON_CLOSE);
        setBounds(100, 100, 800, 500);
        setVisible(true);
    }
}

File: src/MarkdownEditor.kt
import javax.swing.*
import javax.swing.filechooser.FileNameExtensionFilter

import java.awt.Color
import java.awt.Rectangle

import java.io.File

class MarkdownEditor : JFrame(){
    
    private val textArea = JTextArea("(Markdown text will appear here)")
    private val scroll = JScrollPane(textArea)
    private val importButton = JButton("Import File")
    private val saveButton = JButton("Save Changes")
    private val layoutManager = GroupLayout(this.contentPane)
    private lateinit var file: File

    init{
        layoutManager.setHorizontalGroup(layoutManager.createSequentialGroup()
            .addComponent(scroll)
            .addGroup(layoutManager.createParallelGroup(GroupLayout.Alignment.LEADING)
                .addComponent(importButton, 200, GroupLayout.DEFAULT_SIZE, GroupLayout.PREFERRED_SIZE)
                .addComponent(saveButton, 200, GroupLayout.DEFAULT_SIZE, GroupLayout.PREFERRED_SIZE)))

        layoutManager.setVerticalGroup(layoutManager.createParallelGroup(GroupLayout.Alignment.LEADING)
            .addComponent(scroll)
            .addGroup(layoutManager.createSequentialGroup()
                .addComponent(importButton)
                .addComponent(saveButton)))

        importButton.addActionListener {
            val fc = JFileChooser()
            fc.fileFilter = FileNameExtensionFilter("Markdown Files (*.md)", "md")
            fc.isAcceptAllFileFilterUsed = false
            val result = fc.showOpenDialog(null)

            if(result == JFileChooser.APPROVE_OPTION){
                file = fc.selectedFile
                title = file.name + " [MarkdownEditor]"
                textArea.text = file.readText()
                importButton.isEnabled = false
                saveButton.isEnabled = true
                textArea.isEnabled = true
            }
        }

        saveButton.addActionListener {
            file.writeText(textArea.text)
            JOptionPane.showMessageDialog(null, "File successfully saved!")
            title = "MarkdownEditor"
            textArea.text = "(Markdown text will appear here)"
            importButton.isEnabled = true
            saveButton.isEnabled = false
            textArea.isEnabled = false
        }

        scroll.border = BorderFactory.createLineBorder(Color.BLACK)
        textArea.isEnabled = false
        textArea.lineWrap = true
        saveButton.isEnabled = false
        layoutManager.autoCreateGaps = true
        layoutManager.autoCreateContainerGaps = true

        layout = layoutManager
        title = "MarkdownEditor"
        defaultCloseOperation = EXIT_ON_CLOSE
        bounds = Rectangle(100, 100, 800, 500)
        isVisible = true
    }
}
